utils/hct/HashStability.py

```diff
@@ -0,0 +1,146 @@
+                            stderr=subprocess.PIPE,
+                            close_fds=close_fds)
+    stdout, stderr = proc.communicate()
+    res = proc.wait()
+    if res != 0:
+        print(f"normal compile failed {args}, stdout:{stdout}, stderr:{stderr}")
+    return res
+
+def debug_compile(args, output_file, working_dir, empty_env):
+    debug_args = args
+    debug_args.append("-Zi")
+    debug_args.append("-Qstrip_reflect")
+    debug_args.append("-Zsb")
+    debug_args.append("-Fo")
+    debug_args.append(output_file)
+
+    proc = subprocess.Popen(debug_args, cwd=working_dir,
+                            env=empty_env,
+                            # don't writ output to stdout
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            close_fds=close_fds)
+    stdout, stderr = proc.communicate()
+    res = proc.wait()
+    if res != 0:
+        print(f"debug compile failed {args}, stdout:{stdout}, stderr:{stderr}")
+    return res
+
+
+def run_hash_stablity_test(args, dxc_path, dxa_path, test_name, working_dir):
+    empty_env = os.environ.copy()
+    # clear PATH to make sure dxil.dll are not found.
+    empty_env["PATH"] = ""
+    args[0] = dxc_path
+
+    # run normal compile
+    normal_out = os.path.join(working_dir, 'Output', test_name+'.normal.out')
+    res = normal_compile(args, normal_out, working_dir, empty_env)
+    if res != 0:
+        # strip_reflect failed, return fail.
+        return True, "normal compile failed, assume this is an expected failure testing shader."
+
+    normal_hash = extract_hash(dxa_path, normal_out, working_dir, empty_env)
+    if normal_hash is None:
+        return False, "Fail to get hash for normal compilation."
+
+    # run debug compilation
+    debug_out = os.path.join(working_dir, 'Output', test_name+'.dbg.out')
+    res = debug_compile(args, debug_out, working_dir, empty_env)
+    if res != 0:
+        # Zi failed, return fail.
+        return False, "debug compilation failed."
+
+    debug_hash = extract_hash(dxa_path, debug_out, working_dir, empty_env)
+    if debug_hash is None:
+        return False, "Fail to get hash for debug compilation."
+
+    # compare normal_hash and debug_hash.
+    if filecmp.cmp(normal_hash, debug_hash):
+        # hash match, return pass.
+        return True, "Hash matches."
+    else:
+        # hash mismatch
+        return False, "Hash mismatches."
+
+################################################
+################################################
+# For running from the command-line
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Run hash stability test')
+    parser.add_argument('-a','--argument', help='dxc command line options', required=True)
+    parser.add_argument('-p','--path', help='path to the directory containing dxc and dxa', required=True)
+    args = vars(parser.parse_args())
+    dxc_args = args['argument'].split(' ')
```

Comment:
Relying on splitting args as a string is fragile. Instead the "argument" option could be positional and allow multiple values. That would reduce the risk of this causing problems.